### PR TITLE
Update TF S3 backend guide

### PR DESCRIPTION
**Pull request by rrrkharse**
Minor updates to docs:
- Add a link to configuring AWS S3 to backup `.tfstate` to Terraform deployment guides
- Minor rewording in guides

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/docs/deployment/terraform-s3-backend.md [hunk 1]
```diff
@@ -84,4 +84,4 @@ The permissions required by the Terraform user to use AWS S3 as a Terraform back
     EOF
     ```
 
-1. The above configuration will be used the next time Terraform is deployed.
\ No newline at end of file
+1. Deploy Terraform to apply the above configuration.
```
[the remark this answers is not in the document]
Comment (surajkota) [reply website/content/en/docs/deployment/terraform-s3-backend.md:87]: Users now have to checkin this file to their git repo. We should revisit how to configure this next release

**Review by surajkota — CHANGES_REQUESTED** [review]

File: website/content/en/docs/deployment/terraform-s3-backend.md [hunk 2]
```diff
@@ -6,14 +6,20 @@ weight = 70
...
+Having multiple copies of the same `terraform.tfstate` can become difficult to manage and keep in sync between different hosts and users.
+Instead, by using a remote backend, such as AWS S3, state is consolidated in one shared remote location and can be re-used between multiple hosts. Additionally, the state will not be lost if the local `terraform.tfstate` file was accidentally deleted.
 
 For additional details on using AWS S3 as a Terraform backend, refer to the following Terraform [documentation](https://developer.hashicorp.com/terraform/language/settings/backends/s3#s3).
 
+> **Important**
+>
+> Required Terraform variables will still need to be provided as input when deploying even if a remote state file is being used.
+>
+> For example, a cluster name will still need to be provided through the TF_VAR_cluster_name or `.tfvar` file before deploying.
```
Comment (surajkota) [inline website/content/en/docs/deployment/terraform-s3-backend.md:21]: suggest adding another line.

Make sure you provide your `.tfvars` file in the deployment folder and export all the `TF_VAR` variables when making changes to an existing deployment.

File: website/content/en/docs/deployment/terraform-s3-backend.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ The permissions required by the Terraform user to use AWS S3 as a Terraform back
     EOF
     ```
 
-1. The above configuration will be used the next time Terraform is deployed.
+1. Deploy Terraform to apply the above configuration.
```
[the remark this answers is not in the document]
Comment (rrrkharse) [reply website/content/en/docs/deployment/terraform-s3-backend.md]: added, renamed

File: website/content/en/docs/deployment/cognito-rds-s3/guide-terraform.md [hunk 4]
```diff
@@ -105,6 +105,9 @@ pwd
 ### (Optional) Configure Culling for Notebooks
 Enable culling for notebooks by following the [instructions]({{< ref "/docs/deployment/configure-notebook-culling.md#" >}}) in configure culling for notebooks guide.
 
+### (Optional) Configure AWS S3 to backup Terraform state
```
[the remark this answers is not in the document]
Comment (rrrkharse) [reply website/content/en/docs/deployment/cognito-rds-s3/guide-terraform.md]: done

File: website/content/en/docs/deployment/terraform-s3-backend.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rrrkharse) [reply website/content/en/docs/deployment/terraform-s3-backend.md]: reworded, sure

File: website/content/en/docs/deployment/terraform-s3-backend.md (hunk 2 above)
Comment (rrrkharse) [reply website/content/en/docs/deployment/terraform-s3-backend.md:21]: done

**Review by surajkota — APPROVED** [review]

**Pull request merged**